pkg/trace/agent/agent.go

```diff
@@ -36,17 +36,17 @@ const (
 
 // Agent struct holds all the sub-routines structs and make the data flow between them
 type Agent struct {
-	Receiver           *api.HTTPReceiver
-	Concentrator       *stats.Concentrator
-	Blacklister        *filters.Blacklister
-	Replacer           *filters.Replacer
-	ScoreSampler       *Sampler
-	ErrorsScoreSampler *Sampler
-	ExceptionSampler   *sampler.ExceptionSampler
-	PrioritySampler    *Sampler
-	EventProcessor     *event.Processor
-	TraceWriter        *writer.TraceWriter
-	StatsWriter        *writer.StatsWriter
+	Receiver          *api.HTTPReceiver
+	Concentrator      *stats.Concentrator
+	Blacklister       *filters.Blacklister
+	Replacer          *filters.Replacer
+	PrioritySampler   *sampler.PrioritySampler
```

Comment:
👍 Nice, I think removing the `Engine` makes the sampling code really easier to read.